GCC 10 now default to -fno-common which breaks compiling

GCC 10 now defaults to `-fno-common` and it breaks compiling of kabi-dw by default.

https://gcc.gnu.org/gcc-10/changes.html
```
cc -o kabi-dw generate.o ksymtab.o utils.o main.o stack.o objects.o hash.o list.o compare.o show.o parser.yy.o parser.tab.o -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -ldw -lelf
/usr/bin/ld: parser.tab.o:/builddir/build/BUILD/kabi-dw-bd56a6004d5d409d7d03c386400da3f49a8c4c03/parser.h:33: multiple definition of `yyin'; parser.yy.o:(.bss+0x18): first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:72: kabi-dw] Error 1
```
See https://bugzilla.redhat.com/show_bug.cgi?id=1799559

As a workaround, now I need to add `-z muldefs` in LDFLAGS temporary to make GCC happy.

--- a/parser.h
+++ b/parser.h
@@ -30,4 +30,4 @@ int fileno(FILE *stream);
 
 int yylex();
 int yyerror(obj_t **root, char *s);
-FILE *yyin;
+extern FILE *yyin;

